electron-1/preload.js

```diff
@@ -0,0 +1,30 @@
+const { ipcRenderer } = require('electron');
+
+window.bx = {
+  /**
+   * Returns a promise resolving to the actual name of this electron application.
+   * @see https://github.com/electron/electron/blob/master/docs/api/app.md#appgetname
+   */
+  getName() {
+    // Value should come from `electron.app.getName()`
+    return new Promise((resolve) => {
+      ipcRenderer.send('name-message');
+      ipcRenderer.on('name-reply', (event, arg) => {
```

Comment:
It looks to me you are attaching a new callback in any call to `getName` (resp. `getVersion`) without detaching it, no?

This could lead to [EventEmitter memory leak](https://nodejs.org/docs/latest/api/events.html#events_emitter_setmaxlisteners_n), and I guess, you are resolving your promise multiple times.

Reply:
Indeed, so I could use the `.once(...)` method instead of '.on(...)' that would remove the listener next time the event would be invoked.